### PR TITLE
[CLEANUP] - creating test for RepositoriesMeta

### PR DESCRIPTION
**Pull request by kurtwalker**

### CODE REVIEW

**Comment by kurtwalker** [comment]
@pminutillo @bytekast please review

File: engine/test-src/org/pentaho/di/repository/bad-repositories.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (lucboudreau) [inline engine/test-src/org/pentaho/di/repository/bad-repositories.xml]: Should we move the XML files to test-res/ ? Or is it called testfiles/ in kettle?

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  engine/src/org/pentaho/di/repository/RepositoriesMeta.java
  engine/test-src/org/pentaho/di/repository/RepositoriesMetaTest.java
  engine/test-src/org/pentaho/di/repository/bad-repositories.xml
  engine/test-src/org/pentaho/di/repository/repositories.xml

Checkstyle violations:
======================
No errors found

License header violations:
==========================
No violations found








Unit test coverage change:
============
org.pentaho.di.repository.BaseRepositoryMeta:
 * instructionPercentCoverageChange : +35.15%
 * linePercentCoverageChange : +26.83%
 * complexityPercentCoverageChange : +21.43%
 * methodPercentCoverageChange : +25.00%
    
org.pentaho.di.repository.RepositoriesMeta:
 * branchPercentCoverage :68.18%
 * instructionPercentCoverageChange : +88.71%
 * linePercentCoverageChange : +88.87%
 * complexityPercentCoverageChange : +71.77%
 * methodPercentCoverageChange : +96.30%
    
org.pentaho.di.core.lifecycle.KettleLifecycleSupport:
 * branchPercentCoverageChange :**-8.33%**
 * complexityPercentCoverageChange :**-9.09%**
    
org.pentaho.di.core.KettleEnvironment:
 * instructionPercentCoverageChange : +8.28%
 * branchPercentCoverageChange : +11.11%
 * linePercentCoverageChange : +3.23%
 * complexityPercentCoverageChange : +5.26%
 * methodPercentCoverageChange : +10.00%
    
org.pentaho.di.repository.filerep.KettleFileRepositoryMeta:
 * instructionPercentCoverageChange : +67.63%
 * linePercentCoverageChange : +51.52%
 * complexityPercentCoverageChange : +30.77%
 * methodPercentCoverageChange : +30.77%
    
org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * instructionPercentCoverageChange : +0.17%
    
org.pentaho.di.trans.steps.randomvalue.RandomValueMeta:
 * instructionPercentCoverageChange :**-0.15%**
 * branchPercentCoverageChange :**-2.08%**
 * complexityPercentCoverageChange :**-2.13%**
    
org.pentaho.di.trans.step.StepMeta:
 * instructionPercentCoverageChange : +0.19%
 * linePercentCoverageChange : +0.27%
    
org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta:
 * instructionPercentCoverageChange :**-0.18%**
    
org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * instructionPercentCoverageChange :**-0.28%**
 * branchPercentCoverageChange :**-3.33%**
 * complexityPercentCoverageChange :**-1.85%**

File: engine/test-src/org/pentaho/di/repository/bad-repositories.xml (hunk 1 above)
Comment (kurtwalker) [inline engine/test-src/org/pentaho/di/repository/bad-repositories.xml]: Sure, I can do that

Comment (kurtwalker) [inline engine/test-src/org/pentaho/di/repository/bad-repositories.xml]: @lucboudreau actually looks like the build isn't set up for a separate test-res folder

**Comment by lucboudreau** [comment]
Static stuff looks ok. Small drop of coverage in KettleLifecycleSupport. Ready for a proper review. 

cc. @pminutillo @bytekast

**Pull request merged**